pilot/pkg/model/sidecar.go

```diff
@@ -217,7 +217,53 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *Config, configNamespa
 	// Assign namespace dependencies
 	out.namespaceDependencies = make(map[string]struct{})
 	for _, listener := range out.EgressListeners {
-		for _, s := range listener.services {
+		// Explicitly callable services are potential destinations, copy those first
+		serviceDestinations := append([]*Service(nil), listener.services...)
+
+		addDestination := func(d *networking.Destination) {
+			if d == nil {
+				return
+			}
+			// Default to this hostname in our config namespace
+			s, ok := ps.ServiceByHostnameAndNamespace[host.Name(d.Host)][configNamespace]
+			if !ok {
+				// Otherwise pick randomly
+				for _, v := range ps.ServiceByHostnameAndNamespace[host.Name(d.Host)] {
+					s, ok = v, true
```

Comment:
we dont have to cycle through all. Just pick the first one and break

Reply:
I think it could be simiplified like this
```
// Default to this hostname in our config namespace
if s, ok := ps.ServiceByHostnameAndNamespace[host.Name(d.Host)][configNamespace]; ok {
    serviceDestinations = append(serviceDestinations, s)
} else {
    // Otherwise pick the first destination
    for _, v := range ps.ServiceByHostnameAndNamespace[host.Name(d.Host)] {
        serviceDestinations = append(serviceDestinations, v)
        return
    }
}
```

Reply:
And can you please add a unit test with virtual service hosts and ensure it is added in `out.services`?